e2e/tests/node.go

```diff
@@ -181,19 +184,14 @@ func (p *pocketdBin) runCurlCmd(rpcBaseURL, service, method, path, appAddr, data
 		return nil, err
 	}
 
-	// Store the virtual host we want to send in the Host header
-	virtualHost := rpcUrl.Host
-	if len(service) > 0 {
-		// Strip port if it exists and add service prefix
-		host, _, err := net.SplitHostPort(rpcUrl.Host)
-		if err != nil {
-			// No port in the host, use as-is
-			host = rpcUrl.Host
-		}
-		virtualHost = fmt.Sprintf("%s.%s", service, host)
-	}
+	// Get the virtual host that will be sent in the "Host" request header
+	virtualHost := getVirtualHostFromHost(rpcUrl, service)
 
-	// Override the actual connection address if the environment requires it
+	// The current DevNet infrastructure does not support routing requests to arbitrary subdomains.
+	// As a workaround, defaultPathHostOverride (which contains no subdomain) is used
+	// as the gateway's host:port to connect to, and the virtual host (which includes
+	// the service as subdomain) is sent in the "Host" request header.
+	// Override the actual connection address if the environment requires it.
```

Comment:
```suggestion
	// TODO_FUTURE: As of PR #879, DevNet infrastructure does not support routine requests to
	// arbitrary subdomains due to TLS certificate-related complexities.
	// As a workaround,  defaultPathHostOverride (which contains no subdomain) is used as:
	//     1. The gateway's 'host:port' to connect to
        //     2. The virtual host send as part of the "Host" request header (which includes the service as subdomain)
```

Reply:
@red-0ne I still feel like `As a workaround,  defaultPathHostOverride (which contains no subdomain) is used as:`  and `2. The virtual host send as part of the "Host" request header (which includes the service as subdomain)` are contradictory, but I'm not sure what the correct comment i.s